src/pages/index.js

```diff
@@ -41,41 +56,157 @@ const addCardPopup = new PopupWithForm(
+});
 
-const user = new UserInfo(".profile__title", ".profile__description");
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "b7515436-5dfa-4f13-a09a-5183c1df5fb3",
+    "Content-Type": "application/json",
+  },
+});
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 function handleProfileEditSubmit(profileData) {
-  const title = profileData.title;
-  const description = profileData.description;
-  user.setUserInfo(title, description);
-  editProfilePopup.close();
+  console.log("profileData:", profileData);
+  const name = profileData.title;
+  const job = profileData.description;
+  console.log("name:", name, "job:", job); //debugging
+  profileEditSubmitBtn.textContent = "Saving...";
+
+  api
+    .editProfile(profileData.title, profileData.description)
+    .then(() => {
+      user.setUserInfo({ name, job });
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditSubmitBtn.textContent = "Save";
+    });
+}
+
+function handleProfileAddSubmit(newCardData) {
+  profileAddSubmitBtn.textContent = "Saving...";
+  console.log(newCardData); //debugging
+  api
+    .addNewCard(newCardData.title, newCardData.url)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      addCardPopup.close();
+      cardSection.addItem(cardElement);
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileAddSubmitBtn.textContent = "Save";
+    });
 }
 
-function handleProfileAddSubmit(newCardData, cardListEl) {
-  const name = newCardData.title;
-  const alt = newCardData.title;
-  const link = newCardData.url;
-  renderCard({ name, alt, link });
-  addCardPopup.close();
+function handleAvatarEditSubmit(link) {
+  const url = link.url;
+  console.log(url); //debugging
+  profileAvatarSubmitBtn.textContent = "Saving...";
+  api
+    .updateProfilePic(url)
+    .then(() => {
+      user.setUserAvatar(url);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarSubmitBtn.textContent = "Save";
+    });
+}
+
+function handleLikes(cardData) {
+  console.log(cardData); //debugging
+  if (!cardData.like) {
+    //if not liked
+    console.log(cardData.like);
+    api
+      .addLike(cardData._id)
+      .then(() => {
+        cardData.updateLike(true);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else if (cardData.like) {
+    //if liked
+    console.log(cardData.like);
+    api
+      .deleteLike(cardData._id)
+      .then(() => {
+        cardData.updateLike(false);
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors